testing/test_warnings.py

```diff
@@ -562,3 +562,38 @@ def test_hidden_by_system(self, testdir, monkeypatch):
         monkeypatch.setenv(str("PYTHONWARNINGS"), str("once::UserWarning"))
         result = testdir.runpytest_subprocess()
         assert WARNINGS_SUMMARY_HEADER not in result.stdout.str()
+
+
+class TestAssertionWarnings:
+    def test_tuple_warning(self, testdir):
+        testdir.makepyfile(
+            """
+            def test_foo():
+                assert (1,2)
+            """
+        )
+        with pytest.warns(pytest.PytestWarning):
```

Comment:
This will never happen, as `runpytest_subprocess()` runs pytest in a separate process, which won't trigger a `PytestWarning`. You should check `stdout` instead (and no need to run in a separate process, you can use just `runpytest()`.

Reply:
You are right, created #4151 to track this. 👍 

[Here are the full docs to `testdir`](https://docs.pytest.org/en/latest/reference.html#testdir), although they could use some improvement.